### PR TITLE
bottle --merge: check for all cellars

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?
- [ ] Have you successfully run `brew man` locally and committed any changes?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
This fixes (on linux):
```
Error: --keep-old was passed but there are changes in:
cellar: old: "/home/linuxbrew/.linuxbrew/Cellar", new: :"/home/linuxbrew/.linuxbrew/Cellar"
```

**Review by dawidd6 — APPROVED** [review]

File: Library/Homebrew/dev-cmd/bottle.rb [hunk 1]
```diff
@@ -552,7 +552,11 @@ def merge_bottle_spec(old_bottle_spec, new_bottle_hash)
       old_value = old_bottle_spec.send(key)
       next if key == :rebuild && old_value.zero?
       next if key == :prefix && old_value == Homebrew::DEFAULT_PREFIX
-      next if key == :cellar && old_value == Homebrew::DEFAULT_REPOSITORY
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bottle.rb:555]: Would be good to have the same thing for these too: https://github.com/Homebrew/brew/blob/e69a54d0d87889b691ba9d201cb768043af90495/Library/Homebrew/global.rb#L64-L69

Comment (SeekingMeaning) [reply Library/Homebrew/dev-cmd/bottle.rb:555]: Should we use these instead? https://github.com/Homebrew/brew/blob/e69a54d0d87889b691ba9d201cb768043af90495/Library/Homebrew/global.rb#L87-L89

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/bottle.rb:555]: We should check them for every platform so we never set it even if we're running `brew bottle` on a different platform to the platform it's bottling for (e.g. when merging).

Comment (SeekingMeaning) [reply Library/Homebrew/dev-cmd/bottle.rb:555]: Ah, right, I didn't think of that

**Pull request merged**